src/katsdpmetawriter/__init__.py

```diff
@@ -0,0 +1,476 @@
+of the TS (currently supports Redis RDB format).
+
+This writer, when requested, can create two different views of the TS and save these into
+long-term storage.
+
+The first view is a lightweight representation of the TS containing the basic data
+to allow katdal to open an observation. This includes data such as captured timestamps,
+storage configuration information and observation targets. Using the supplied capture block
+ID, an attempt is also made to only record meta information specific to that capture
+block ID.
+
+The second is a complete dump of the entire TS. This may contain meta-data from
+other capture sessions.
+"""
+
+import logging
+import os
+import socket
+import asyncio
+import time
+import enum
+import pathlib
+from collections import deque
+
+import boto
+import boto.s3.connection
+import katsdptelstate
+from katsdptelstate.rdb_writer import RDBWriter
+from aiokatcp import DeviceServer, Sensor, FailReply
+
+
+# BEGIN VERSION CHECK
+# Get package version when locally imported from repo or via -e develop install
+try:
+    import katversion as _katversion
+except ImportError:
+    import time as _time
+    __version__ = "0.0+unknown.{}".format(_time.strftime('%Y%m%d%H%M'))
+else:
+    __version__ = _katversion.get_version(__path__[0])    # type: ignore
+# END VERSION CHECK
+
+
+logger = logging.getLogger(__name__)
+
+# Fairly arbitrary limit on number of concurrent meta data writes
+# that we allow. Tradeoff between not stopping observations and
+# taking too long to discover some blocking fault.
+MAX_ASYNC_TASKS = 10
+
+# Template of key names that we would like to preserve when dumping
+# a lite version of Telstate. Since we always back observations with
+# a full dump of Telstate, we don't fail on missing entries, but do
+# log them.
+# Each string entry is format'ed with the following substitutions:
+#
+#    {cb}: The capture block ID which uniquely identifies this data capture
+#    {sn}: The name of a specific stream in the capture (e.g. sdp_l0)
+#    m???: A special that represents a glob-style pattern match used to
+#          do wildcard MeerKAT antenna name matching for certain keys.
+#    s????: A special that represents a glob-style pattern match used to
+#          do wildcard SKA prototype antenna name matching for certain keys.
+#
+LITE_KEYS = [
+    "{sn}_int_time",
+    "{sn}_sync_time",
+    "{sn}_bls_ordering",
+    "{sn}_n_chans",
+    "{sn}_bandwidth",
+    "{sn}_center_freq",
+    "{sn}_s3_endpoint_url",
+    "{sn}_stream_type",
+    "{sn}_need_weights_power_scale",
+    "{cb}_obs_params",
+    "{cb}_obs_script_log",
+    "{cb}_obs_label",
+    "{cb}_obs_activity",
+    "{cb}_{sn}_chunk_info",
+    "{cb}_{sn}_first_timestamp",
+    "sub_pool_resources",
+    "sub_band",
+    "sub_product",
+    "m???_observer",
+    "m???_activity",
+    "m???_target",
+    "s????_observer",
+    "s????_activity",
+    "s????_target",
+    "cbf_target",
+    "sdp_config"
+]
+
+
+def timer():
```

Comment:
So how does this actually help vs time.time() ? Can't that be mocked as well if needed ?

Reply:
I ran into issues because the event loop uses time.monotonic internally, which disrupted my attempts to carefully feed specific return values to specific calls. With a wrapper I can not just mock the calls I want, but mock *only* the calls I want.